File: ruby/hamming/hamming.rb
class Hamming
  VERSION = 1

  def self.compute(s1, s2)
    raise ArgumentError if s1.length != s2.length
    sc1 = s1.chars
    sc2 = s2.chars

    count = 0
    (0...sc1.length).each do |i|
      count += 1 if sc1[i] != sc2[i]
    end
    count

  end
end

File: javascript/hamming/hamming.js
var Hamming = function () {}

Hamming.prototype.compute = function (strand1, strand2) {
  var count = 0;
  if (strand1.length !== strand2.length) {
    throw new Error('DNA strands must be of equal length.')
  } else {
    for (i = 0; i < strand1.length; i++) {
      if (strand1[i] !== strand2[i]) {
        count++;
      }
    }
  }
  return count;
}

module.exports = Hamming
